### PR TITLE
[Core] Adding OrthonormalBasis to MathUtils

### PR DESCRIPTION
**Pull request by loumalouomega**

### CODE REVIEW

File: kratos/utilities/math_utils.h [hunk 1]
```diff
@@ -982,6 +982,30 @@ class MathUtils
...
 
+    /**
+     * @brief This computes a orthonormal basis from a given vector
+     * @param a First resulting vector
+     * @param b Second resulting vector
+     * @param c The input vector
+     * @note Orthonormal basis taken from: http://orbit.dtu.dk/files/126824972/onb_frisvad_jgt2012_v2.pdf
+     */
+    template< class T1, class T2 , class T3>
+    static inline void OrthonormalBasis(const T1& c,T2& a,T3& b ){
```
Comment (jcotela) [inline kratos/utilities/math_utils.h:993]: Note that the params above are not in order.

**Review by jcotela — APPROVED** [review]

File: kratos/utilities/math_utils.h (hunk 1 above)
Comment (loumalouomega) [reply kratos/utilities/math_utils.h:993]: We can live with that :P

File: kratos/utilities/math_utils.h [hunk 2]
```diff
@@ -982,6 +982,30 @@ class MathUtils
...
+        KRATOS_DEBUG_ERROR_IF_NOT(norm_2(c) < 1.0) << "Input should be a normal vector" << std::endl;
```
Comment (jcotela) [inline kratos/utilities/math_utils.h:994]: @loumalouomega this check is making the test fail when I compile kratos in debug mode (the input has norm == 1). Are you sure that this is the right check?

Comment (loumalouomega) [reply kratos/utilities/math_utils.h:994]: Huuum, could a problem of tolerance, I can add a less restrict check (BTW if the norm is greater than one will cause errors)

Comment (jcotela) [reply kratos/utilities/math_utils.h:994]: well, if this is a problem maybe you need to change the test. The test input is `[0.0, 1.0, 0.0]`.

Comment (loumalouomega) [reply kratos/utilities/math_utils.h:994]: The norm is 1, it is supposed to not give an error

**Comment by pooyan-dadvand** [comment]
Is it working now in debug?

**Comment by loumalouomega** [comment]
@RiccardoRossi ask me to do some modifications, pushing in some minutes

**Pull request merged**